### ISSUE TITLE
[Android Bootcamp] Project 1 - Review my InstagramClient app

### ISSUE THREAD

**Issue by lita** [lita/InstagramClient#6]
/cc @codepathreview @codepath

I want to try to work on this more Sunday, but I am submitting early just in case. Would it be okay if I append to this issue if I do further work?

Once I added the video feature to the main app, the stream became clunky and slow. I want to change it so that clicking on the video will create a new screen of the full-size video playing.

**Comment by nesquena** [comment]
Yes, if you do further work, before Sunday at 10, you can update the README and reply to this issue to notify us. Any additional work to improve the UI / UX is encouraged.

**Comment by lita** [comment]
I know it is late, but I did manage add sticky headers and make the video play full screen. Would have submitted sooner but didn't get internet until now. I hope you can review the extra work I did up till now!

**Comment by codepathreview** [comment]
Excellent work, Lita. A few notes after checking out the code:
- Good layout, proper view naming, usage of sp/dp, margin and relative placement rules
- Good polished UI and correct use of Styles, Colors and String resources
- Nice to see you used the SwipeRefreshLayout for easy stream refreshes
- Good job using the ViewHolder pattern to improve performance of photos ListView.
- Nice to see you got the full screen video working.
- Nice to see the sticky header.
- Consider better organizing your source files
- Consider notifying the user in onFailure method.
- Consider checking for internet connection before making a network call.

We have provided a detailed Project 1 Feedback Guide here which covers the most common points we see for this project. Read through the feedback guide point-by-point to determine other ways you could improve your submission. You should consider going back and implementing applicable feedback as well. Keep in mind that one of the most important parts of Android development is learning the correct patterns and conventions.

Hopefully this first project has given you a better sense of working with RelativeLayout which is a very flexible layout system, probably one of the most powerful responsive-first layout systems available across web and mobile platforms. This assignment also gave us our first introduction to networking, working with APIs and loading remote images. The next assignment will introduce new concept such as accepting user input and navigating between activities but will also reinforce important concepts such as networking, using APIs, handling remote images.

If you have any particular questions about the assignment in general or on any of the feedback, feel free to reply here or or email us support@codepath.com.

P.S. Good to see you properly added the readme with features and screenshots to your project as described in the submitting assignments guide!

**Comment by lita** [comment]
Thank you for the feedback!

**Comment by lita** [comment]
I would like to learn more about organizing your source files, as I had trouble with this. Can you provide further suggestions on how to go about organizing my source files?

**Comment by nesquena** [comment]
You can take a look at http://guides.codepath.com/android/Organizing-your-Source-Files for a good starting point!
